docs/entities/serviceclaim.md

```diff
@@ -0,0 +1,49 @@
+# ServiceClaim
+
+A Service Claim represents a claim for Registered Service.
+
+## Specification
+
+The definition of a ServiceClaim can be obtained directly from our [ServiceClaim
+CRD](../../config/crd/bases/primaza.io_serviceclaims.yaml). The specification
+contains two required properties:
+
+- ServiceClassIdentity: A set of key/value pairs that identify the service
+  class. Examples of service class identity keys include type of service, and
+  provider of service. This property is required.
+- ServiceEndpointDefinitionKeys: An array of keys that is required for
+  connectivity. The values corresponding to each of these keys will be extracted
+  from the service. This property is required.
+
+There are other three optional fields. The EnvironmentTag and
+ApplicationClusterContext are mutually exclusive:
+
+- Application: Fields indentifies application resources through kind, apiVersion and label selector. 
+- EnvironmentTag: A string representing one of the environment.
+- ApplicationClusterContext: A combination of environment name and namespaces.
+
+## Status
+
+The Status of the ServiceClaim is also defined under the [ServiceClaim
+CRD](../../config/crd/bases/primaza.io_serviceclaims.yaml). It contains only one
+required property to track the state. The state could be either `Pending` or
+`Resolved`. If the state is `Resolved`, there should be Secret and Service
+Binding resources created
+
+There is an optional `claimID` field with a unique ID for the claim.  And there is
+another optional field, `registeredService` that points the service.
+
+## Use Cases
+
+### Creation
+
+When a Service Claim is created, Primaza should find a Registered Service based on Service Class Identity and Service Endpoint Definition Keys and create Secret and Service Binding resources. Then it should update the state accordingly.
+
+### Deletion
+
+When a Service Claim is deleted, Primaza should delete the Service Endpoint Definition Secret and the Service Binding.
```

Comment:
It should also change the state of the Registered Service

Reply:
Done